### DISCUSSION TITLE
Data source to support tables with a non uuid partition key

### DISCUSSION THREAD

**Discussion by mshanmugam** [HadesArchitect/GrafanaCassandraDatasource#79] [Q&A] [answered]
Hi how do I enable tables that has time series data but with partition key that is non uuid/ time based.

**Comment by HadesArchitect** [comment, marked as answer]
Hi! Could you give an example? Like `create table... `

**Reply by mshanmugam** [reply]
For example

create table employee ( emp_id text, emp_name text, login_time timestamp,  PRIMARY KEY  emp_id)


If I want to build report on emp_id with a timeseries of login_time , how will I do that with the current plugin version . 
Since current plugin expects partition to be timestamp based uuid.

**Reply by HadesArchitect** [reply]
I see. It should be doable using query editor. I will check it later during the weekend.

**Reply by HadesArchitect** [reply]
**Data**
```
create keyspace test with replication = {'class': 'SimpleStrategy', 'replication_factor': 1};
create table test.employee ( emp_id int, emp_name text, login_time timestamp, PRIMARY KEY (emp_id));
insert into test.employee (emp_id, emp_name, login_time) values (1, 'Empl1', '2021-02-14T18:54:13+00:00');
insert into test.employee (emp_id, emp_name, login_time) values (2, 'Empl2', '2021-02-14T17:54:13+00:00');
insert into test.employee (emp_id, emp_name, login_time) values (3, 'Empl3', '2021-02-14T16:54:13+00:00');
```

**Query Editor**
`SELECT emp_name, CAST(emp_id as double), login_time FROM test.employee WHERE emp_id IN (1, 2, 3) AND login_time > $__timeFrom and login_time < $__timeTo allow filtering`

**Result**
![image](https://user-images.githubusercontent.com/1742301/107887051-bd830f00-6f03-11eb-8f64-00f1e9aee1b6.png)

**Reply by HadesArchitect** [reply]
So, there are some points:

1. Statement "plugin expects partition to be timestamp based uuid" is not correct. The plugin can work with any valid partitions
2. I don't understand what exactly you would like to visualize. Every timeseries data consists of three components: a source ID, a timestamp, a value. In your example, the source ID is an employee ID, the timestamp is login_time, but what is the value?

I'll be glad to help but I need you to clarify the use-case. What is the final goal here? How the visualization should look like?

**Reply by mshanmugam** [reply]
Thanks. Appreciated the response. How do i access query editor directly.  Also we want to build a timeseries based on the user login pattern. how frequently a set of users login etc.  Sorry couldn't respond quick since we were in the middle of a snow storm.

**Reply by HadesArchitect** [reply]
@mshanmugam You can access query editor [as explained](https://github.com/HadesArchitect/GrafanaCassandraDatasource#query-editor) in the docs.

If you want to build a time-series based on user login events, you need another data model table. In your example, the primary key consists of the employee id only, which is not enough: due to upsert, every next login will overwrite the previous one. 

If you could give the exact picture of what you want to see in grafana, I'll be glad to help.
